### ISSUE TITLE
`knitr::knit()` evaluates chunk if false value of `eval` chunk option has a comment.

### ISSUE THREAD

**Issue by nplatonov** [yihui/knitr#2199]
Input:

` ```{r}`
`evaluate <- FALSE`
`comment(evaluate) <- "not today"`
` ``` `

` ```{r, eval=evaluate}`
` str(knitr::opts_current$get("eval"))` 
` ``` `

` ```{r} `
` xfun::session_info('knitr') ` 
` ``` `

Output:

```r
evaluate <- FALSE
comment(evaluate) <- "not today"
```

```r
str(knitr::opts_current$get("eval"))
```

```
##  logi FALSE
##  - attr(*, "comment")= chr "not today"
```

```r
xfun::session_info('knitr')
```

```
## R version 4.2.2 Patched (2022-11-17 r83369 ucrt)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19044)
## 
## Locale:
##   LC_COLLATE=English_United Kingdom.utf8 
##   LC_CTYPE=English_United Kingdom.utf8   
##   LC_MONETARY=English_United Kingdom.utf8
##   LC_NUMERIC=C                           
##   LC_TIME=English_United Kingdom.utf8    
## 
## Package version:
##   evaluate_0.18   glue_1.6.2      graphics_4.2.2  grDevices_4.2.2
##   highr_0.9       knitr_1.40      magrittr_2.0.3  methods_4.2.2  
##   stats_4.2.2     stringi_1.7.8   stringr_1.4.1   tools_4.2.2    
##   utils_4.2.2     xfun_0.34       yaml_2.3.6
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('knitr')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/knitr')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
Thanks for the report. This is because we are using `xfun::ifFALSE()` to check the `eval` option, and it does not account for attributes

``` r
evaluate <- FALSE
comment(evaluate) <- "not today"
xfun::isFALSE(evaluate)
#> [1] FALSE
base::isFALSE(evaluate)
#> [1] TRUE
```

**Comment by cderv** [comment]
@yihui I made a PR in case you think make `xfun::isFALSE()` same as `base::isFALSE()` a good idea. 
Otherwise, we just need to account for attribute in `identical()`

**Comment by yihui** [comment]
This should be fixed by ccb07c4c2266666f7375e5df242103672bd629ac, which has been included in the latest CRAN release (v1.42). Thanks!
